Objects/descrobject.c

```diff
@@ -314,21 +314,20 @@ classmethoddescr_call(PyMethodDescrObject *descr, PyObject *args,
     self = PyTuple_GET_ITEM(args, 0);
     if (!PyType_Check(self)) {
         PyErr_Format(PyExc_TypeError,
-                     "descriptor '%V' requires a type "
-                     "but received a '%.100s'",
+                     "descriptor '%V' requires a subtype of '%.100s' "
```

Comment:
I am not sure that `subtype of '%.100s'` is better than just `type`. It contains more information, but I think that this information is redundant at that moment and distracts from the true error -- passing an instance instead of a type.

Maybe just remove `PyDescr_TYPE(descr)->tp_name`?